re/ReForm_Validation.re

```diff
@@ -3,25 +3,59 @@
 module I18n = {
   type dictionary = {
     required: string,
-    email: string
+    email: string,
   };
   let ptBR = {
     required: {j|Campo obrigatório|j},
-    email: {j|Email inválido|j}
+    email: {j|Email inválido|j},
   };
   let en = {required: "Field is required", email: "Invalid email"};
 };
 
+type error = option(string);
+
+type errors = option(list(string));
+
 type validation('values) =
   | Required
   | Email
-  | Custom('values => option(string));
+  | Custom('values => error);
+
+/* turn list(error) => errors */
+let squeeze = (errorList: list(error)) => {
+  let errorList_ = errorList |> List.filter(error => error != None);
+  List.length(errorList_) > 0 ?
+    Some(
+      errorList_
+      |> List.map(opt =>
+           switch (opt) {
+           | Some(msg) => msg
+           | None => ""
+           }
+         ),
+    ) :
+    None;
+};
 
 let getValidationError =
-    ((_, validator), ~values, ~value, ~i18n: I18n.dictionary) =>
-  switch validator {
-  | Required => String.length(value) < 1 ? Some(i18n.required) : None
+    (validator, ~values, ~value, ~i18n: I18n.dictionary)
+    : error =>
+  switch (validator) {
+  | Required => String.length(value) > 0 ? None : Some(i18n.required)
   | Custom(fn) => fn(values)
   | Email =>
-    Js.Re.test(value, [%bs.re {|/\S+@\S+\.\S+/|}]) ? None : Some(i18n.email)
+    String.length(value) == 0
```

Comment:
I think `String.length(value) == 0` should be kept out, if the string is empty it is indeed an error

Reply:
how to deal with this situation? the field is optional, but if it's not empty string, it should be a valid email ?

Reply:
Ah, I got your point hmm I think we can let like that indeed for now